### PR TITLE
Fix error `InvalidMavenPublicationException`

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Fix error `org.gradle.api.publish.maven.InvalidMavenPublicationException:` 
Coming from the comment https://github.com/opensearch-project/notifications/pull/549#issuecomment-1278186297

### Issues Resolved
https://github.com/opensearch-project/notifications/issues/501

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Build pass on my local with this change 
`./build.sh manifests/2.4.0/opensearch-2.4.0.yml --component notifications-core`, `./build.sh manifests/2.4.0/opensearch-2.4.0.yml --component notifications`

maven folder with `org.opensearch.plugin`
```
pwd
/usr/share/opensearch/git/opensearch-build/tar/builds/opensearch/maven/org/opensearch/plugin
[opensearch@ee8276647b38 plugin]$ ls -ltr
total 8
drwxrwxr-x 3 opensearch opensearch 4096 Oct 13 23:49 notifications
drwxrwxr-x 3 opensearch opensearch 4096 Oct 13 23:50 opensearch-notifications-core
```

**Review by lezzago — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]

**Comment by lukas-vlcek** [comment]
@prudhvigodithi This is interesting.

Right now I am not sure why this is happening. But I noticed one interesting this, without your modification from this PR I am still able to build both plugins individually like this:

```shell
# Let's get to root notification folder which hosts subprojects (the plugins)
$ cd notifications

# Start with the core plugin
$ cd core
$ ../gradlew clean publishToMavenLocal

# Switch to notification plugin and build it
$ cd ../notifications
$ ../gradlew clean publishToMavenLocal
```

And now I can see the following in my local mvn repo
```shell
$ pwd
/Users/lukas.vlcek/.m2/repository/org/opensearch/plugin

$ ls -ltr
drwxr-xr-x  4 lukas.vlcek  staff  128 Oct 14 20:57 opensearch-notifications-core
drwxr-xr-x  4 lukas.vlcek  staff  128 Oct 14 20:59 notifications
```
So both the plugins were build correctly (at least it seems to me).
Except I am not just sure why the notifications plugin is deployed into `notifications` folder and not into `opensearch-notifications` folder, but this might be different issue.

It feels to me that something is stepping on each other's toes when gradle is run from the parent folder (which I think is the case of the `opensearch-build` script).

**Comment by prudhvigodithi** [comment]
Hey @lukas-vlcek thanks, if so we should consider modifying the [notifications-core](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/components/notifications-core/build.sh) and [notifications](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/components/notifications/build.sh) build scripts and execute the [build workflow](https://github.com/opensearch-project/opensearch-build/tree/main/src/build_workflow)
Example:
`./build.sh manifests/2.4.0/opensearch-2.4.0.yml --component notifications-core`
`./build.sh manifests/2.4.0/opensearch-2.4.0.yml --component notifications`

Once this build successfully we can then remove the added setting `startParameter.excludedTaskNames=["publishPluginZipPublicationToMavenLocal"]` wdyt?
Thank you

**Comment by lukas-vlcek** [comment]
The is also a nebule plugin upgrade PR going on https://github.com/opensearch-project/OpenSearch/pull/4810 (it might be relevant, or not... I do not know atm). Let's keep a reference this PR here (even if it is for `main`).

**Comment by lukas-vlcek** [comment]
I had a short chat with @prudhvigodithi about this yesterday and the following is a summary:

TL;DR: We are ok for now to move forward with this PR as it is.

For now **we are ok with the workaround provided in this PR** as it is unblocking the build process. The side-effect of this PR is that OpenSearch plugins (`opensearch.opensearchplugin`) are not built for the `publishToMavenLocal` task but they are not needed for the local repo anyway (unlike to the staging repo), so this is acceptable compromise.

It would make sense to try to run the build for each individual plugin as suggested and demonstrated above because this does not lead to the exception and all plugins are built as expected. But that would require changes to the build scripts itself and that can be a risk that we do not want to take now given [we are approaching `2.4` code freeze soon](https://opensearch.org/releases.html).
